### ISSUE TITLE
NO VOICE

### ISSUE THREAD

**Issue by toxmost** [Antox/Antox#409]
LG Nexus 5x  Bullhead 7.1.1
======================
No audio in call - Antox,
video - picture rotate bag...
Sound go to one side: Antox_v0.25 ----> Antox (old) - ok
Antox (old) ---->Antox_v0.25 - no sound
Antox_v0.25 <----> Antox_v0.25  - no sound

**Comment by anon8675309** [comment]
This appears to happen on multiple devices.  I can not hear other people on any of my Android devices (running Moto G w/ Android 5.1.1, Samsung Tab A w/ Android 6,0.1 and Nexus 6P w/ Android 7.1.1), however a friend of mine can hear me crystal clear on his Nexus tablet (I'll try to find out what version of Android he's running).  All the devices are running Antox 0.25.456 (15456) from the Play store, all permissions are enabled, and it doesn't matter if he calls me or I call him.  I've also tried an audio call with Antidote (the iOS client) and I can not hear them nor can they hear me.

Aside from getting his Android version, what further information can I gather to help troubleshoot this?

**Comment by toxmost** [comment]
I think it's in the reception and decoding of packets, which means in the nucleus TOX, because as a result of experiments, if a long time to speak to both the phone's voice appears on the second and then disappears. Judging by the code of big changes EXCEPT FOR THE TRANSITION TO NEW CORE - the program did not only cosmetics mostly.

**Comment by toxmost** [comment]
How investigation?

**Comment by zoff99** [comment]
the problem is that on your phone with 3g internet mostly are using TCP

**Comment by toxmost** [comment]
It does not matter what kind of internet is now working - 3g or WiFi, the result is the same.
See https://github.com/Antox/Antox/issues/409 <https://github.com/Antox/Antox/issues/409>




> 14 февр. 2017 г., в 18:45, Zoff <notifications@github.com> написал(а):
> 
> the problem is that on your phone with 3g internet mostly are using TCP

**Comment by zoff99** [comment]
how do you know if you are using UDP or TCP?
what version Antox do you have installed?
are you trying to do Antox <-> Antox? or Antox <-> uTox/qTox ?
voice always takes a few seconds to start up

**Comment by toxmost** [comment]
how do you know if you are using UDP or TCP?
— To settings have «Enable UDP» checkbox, i try
===================
what version Antox do you have installed?
— Antox 0.24.428. work audio OK
===================
are you trying to do Antox <-> Antox? or Antox <-> uTox / qTox?
— I try to uTox (os x) and Antidote (ios) - similar
===================
voice always takes a few seconds to start up
— I wait and talk 1 minute - over 13-16 sek i listen a piece of my voice (>1sek), and further silence.

**Comment by oouuppss** [comment]
hi, first try today with Tox protocol
latest antox from playstore, mobile htc 816 desire, updated with marshmallow 6.0.
qtox pc calling me:    qtox can ear me but i got no sound from him
i call qtox:  he can ear me, but no sound too for me, and i got sound icon checked like close
Ty.

**Comment by wiiaam** [comment]
Will investigate soon, can you guys confirm that this is only happening in the recent updates?

**Comment by toxmost** [comment]
confirm, that this is only happening in the recent update.

**Comment by wiiaam** [comment]
Can you hear audio on either end of the call?

**Comment by toxmost** [comment]
no, only one side, if i use Antox less 25 version in one side and larger v.25 on the other side . If use larger v.25 in two side - no audio.

**Comment by wiiaam** [comment]
What side doesn't have audio when you are using 0.24 - 0.25?

**Comment by toxmost** [comment]
Sound go to one side: Antox_v0.25 **---->** Antox (24) or Antidote or uTox

**Comment by toxmost** [comment]
If i talk 1 minute in Antox (24) or Antidote or uTox **---->** Antox_v0.25   - over 13-16 sek i listen a piece of my voice (>1sek), and further silence.

**Comment by toxmost** [comment]
How investigation?

**Comment by toxmost** [comment]
Oh.....I think everyone does not care....):

**Comment by Sheogorath2** [comment]
Elephone P8000, android 5.1. No output sound in antox, input/output in qtox fine while doing antox <-> qtox call or vice versa. In antox the speaker icon is crossed out. When starting a video call or enabling camera the crossed speaker icon turns to phone with two arcs above it, clicking it crashes the app.

**Comment by zoff99** [comment]
@shantanu1227 can you attach the crashlog please?

**Comment by Maestroxr** [comment]
> Elephone P8000, android 5.1. No output sound in antox, input/output in qtox fine while doing an antox <-> qtox call or vice versa. In antox the speaker icon is crossed out. When starting a video call or enabling camera the crossed speaker icon turns to phone with two arcs above it, tapping it crashes the app.

I can confirm the above behaviour, got have that happen on my galaxy s3 too.

I have also tried several times antox<->antox  didn't work for me.
 while utox<->antox worked one way, I could talk to the android phone and hear it on my pc, but not hear my pc's mic input on my phone.
I'm using Galaxy S3 android 4.3

**Comment by smartes** [comment]
Confirm on utox<-antox. I can hear on Utox (ubuntu), but not hear on antox Galaxy S5 android 6.0.1.

**Comment by zoff99** [comment]
@smartes if Antox crashes, please attach the crashlog here.
the latest version should save the crashlog on the sd-card

(!! be careful it may contain private information !!)

**Comment by Sheogorath2** [comment]
Sorry for late response. Here's my logs:
https://gist.github.com/Sheogorath2/c2b6ceeecd51daa265e16e4032b4e559
Antox 0.25.476 (15476) from google play on Android 5.1, Elephone p8000

**Comment by zoff99** [comment]
hmm ok if fails at this call:
https://github.com/TokTok/c-toxcore/blob/master/toxav/toxav.api.h#L378

but i can only fix it in my local repo. for a real fix @wiiaam needs to check this

**Comment by vitja2002** [comment]
Could you please share the fixes.

**Comment by marmistrz** [comment]
I can reproduce it on every Android device I use. The desktop, qTox, receives the sound from the phone's microphone, but the phone (Antox) gives no sound and is stuck on `ringing`. Nothing interesting in logcat. 

Samsung Galaxy J7 and OnePlus 3 were tested.

**Comment by zoff99** [comment]
i was referring to: *And, also, antox may crash with "no reason".* from @smartes 
please attach the crashlogs

**Comment by smartes** [comment]
Sorry for delay, here a part of my crashlog.
[crash_2017-04-13_14_21_01.txt](https://github.com/Antox/Antox/files/925211/crash_2017-04-13_14_21_01.txt)

**Comment by zoff99** [comment]
@smartes could be a problem with nodes file. do you have some sort of firewall on your phone? or xposed?

**Comment by zoff99** [comment]
@smartes try if it works without crash on this version:
https://circleci.com/api/v1/project/zoff99/Antox/latest/artifacts/0/$CIRCLE_ARTIFACTS/Antox.apk?filter=successful&branch=zoff99%2FAntox_v0.25.2_fullsource

!! use at your own risk !!

**Comment by Sheogorath2** [comment]
@zoff99 no luck. Same crashes. I have xposed.

**Comment by zoff99** [comment]
@Sheogorath2 yeah, i see your crash log shows its crashing amidst calls. my version will only fix the missing bootstrap nodes data.
if i have time i can put some fixes in for the call crashes ...

**Comment by 0nomatopoeia** [comment]
@zoff99 can you upload 0.24 build with your bootstrap nodes fix?
incoming audio is broken in 0.25 (i'm using your 0.25.1 build & only outgoing audio works)

**Comment by zoff99** [comment]
@0nomatopoeia do you have any idea where audio is broken? what commit did it?
i am trying to revert to an older c-toxcore to fix it.
(0.24 can't be used, since it 100% crashes on many devices.)

**Comment by 0nomatopoeia** [comment]
@zoff99 i'm using this version: https://circleci.com/api/v1/project/zoff99/Antox/latest/artifacts/0/$CIRCLE_ARTIFACTS/Antox.apk?filter=successful&branch=zoff99%2FAntox_v0.25.1 (the f-droid repo build kept crashing)
where can i find the commit hash?
and how can i determine where the audio is broken?
also, perhaps 0.24 crashes because of the bootstrap nodes issue?

**Comment by zoff99** [comment]
forget about 0.24.
i am having issues with compiling now. somebody changed the libsodium git repo :-(
**grrrrrr**

**Comment by zoff99** [comment]
@0nomatopoeia can you help with development? or testing?
what kind of skills do you have? any help would be welcome ...

**Comment by 0nomatopoeia** [comment]
@zoff99 i can help with testing.
my coding is not yet up to snuff i'm afraid [º﹏º]

**Comment by zoff99** [comment]
@0nomatopoeia can you try out this version:
https://622-57031623-gh.circle-artifacts.com/0/tmp/circle-artifacts.zCXoevM/Antox_circleci_abd0b705604a5480d28531c74cc9f138459d9af2.apk

it's reverted back to c-toxcore 0.1.5
let's see if there is any difference

**Comment by 0nomatopoeia** [comment]
@zoff99 no difference, both outgoing audio & video work fine, no incoming

**Comment by zoff99** [comment]
@0nomatopoeia yeah, it's was i thought would happen.
i will do a reference implementation on android. if that is ready maybe you can test it?
stay tuned ...

**Comment by zoff99** [comment]
@0nomatopoeia is you android device on 3g (or wlan)?
can you test how long it takes for the other party to see your video , when your android device is connected via 3g (NATed)

**Comment by 0nomatopoeia** [comment]
@zoff99 until now i've just been using wlan
via 3g it took 2-4 seconds (3 times 2 seconds, once 4 seconds) to establish the call (between the dial tone started on antox & it started ringing on qtox). and 8 seconds for qtox to start receiving video the first time, about 2 seconds all subsequent times

**Comment by nazar-pc** [comment]
Same here on Android 7.1.2 (Lineage OS)@Nexus 6p.
I can hear well on desktop side (uTox/qTox on Linux), but can't hear anything on mobile.

**Comment by anon8675309** [comment]
I have a test device at home that I can use to try and track down this
issue. I don't know how to build to-do, but I can install the apk listed
earlier in this thread. I won't be home until next week, so maybe someone
will beat me to the punch.

**Comment by Mutisk** [comment]
It seems to affect me using Moto G4 / Android 7. Speaker icon crossed over after established call (touching it does nothing) but remote part can hear me well.

**Comment by NishaYume** [comment]
Same problem , no sound, and mega super pixelated video chat,unusable for a Nubia n1 black. Android v 6.0. :(.

**Comment by zoff99** [comment]
@NishaYume if you want you can try this:
https://github.com/zoff99/ToxAndroidRefImpl/tree/zoff99/dev003

but its really early early alpha!

**Comment by subliun** [comment]
I think I've fixed this
